stock_inventory/models/stock_inventory.py

```diff
@@ -191,24 +191,56 @@ def _get_domain_category_quants(self, base_domain):
             ]
         )
 
+    def _get_products(self):
+        self.ensure_one()
+        if self.product_selection == "all":
+            return self.env["product.product"].search([])
+        elif self.product_selection == "category" and self.category_id:
+            return self.env["product.product"].search(
+                [("categ_id", "=", self.category_id.id)]
+            )
+        else:
+            return self.product_ids
+
+    def _get_overlapping_inventories(self):
+        for rec in self:
```

Comment:
here it should be only one record actually, right?

Reply:
yes, indeed, I had self.ensure_one() before but sometimes it was called with an empty recordset and that was causing tests to fail

Reply:
yes I saw that, for some reason in the test of stock_cycle_count they are calling the method with empty self. I think the problem in this case is with the test, I don't see the reason why it should be done that way

Reply:
perhaps that test is incorrect, but I am not sure if the method it can be called with empty recordset in any other way, and I thought that doing this change was safer

Reply:
The tests for the stock_cycle_count are incorrect as far as I can see, but the fix is not totally clear to me, I will try to investigate a bit more